### DISCUSSION TITLE
What to choose? OLM or operator-controller?

### DISCUSSION THREAD

**Discussion by TheDukeDK** [operator-framework/operator-controller#935] [Q&A] [answered]
We are super familiar with Helm and use several operators which we install with Helm. 

However, we are now evaluating Infinispan as a cache on Kubernetes and it doesn't provide a Helm chart for installing the operator. 

Instead the preferred approach seems to be with OLM from OperatorHub.io. But the OLM repos says it is in maintenance mode and refers people here.

Whats the advice here. Just use the OLM approach until it is deprecated?

**Comment by camilamacedo86** [comment, marked as answer]
Hi @TheDukeDK,

For distributing your project on OperatorHub.io, you will continue doing everything the same way for now—nothing changes. In the future, you might have the option to push a HelmChart instead of needing to create an Operator with a bundle CSV for OLM.

### Key Points to Note:
- Both OLM and operator-controller manage operators and primarily handle similar projects.
- **OLM** is the deprecated project (version v0).
- **operator-controller** is the newer solution (version v1), which is nearing a stable release (v1.0.0) but is not there yet.

**In other words, operator-controller is the new version of OLM.**

Both solutions currently accept the CSV bundle format. The motivations for moving to v1 are outlined here: [OLMv1 Design Decisions](https://operator-framework.github.io/operator-controller/project/olmv1_design_decisions/).

Let me know if you have any questions!

**Reply by camilamacedo86** [reply]
Important, currently OLMV1 can work with a subset of features of OLMV0; for more information regarding the current limitations of OLMv1, see [limitations](https://github.com/camilamacedo86/operator-controller/blob/e7ee00514b34d2e75975cb2d63775a0595fbdc48/docs/project/olmv1_limitations.md).

I recommend you checking [the conceptual differences between OLMv0 and OLMv1](https://github.com/camilamacedo86/operator-controller/blob/e7ee00514b34d2e75975cb2d63775a0595fbdc48/docs/project/olmv1_design_decisions.md), you will see that one of the motivations is to accept content in other formats like Helm Chart, but for no,w OLMV1 works with bundle-registry:v1 with some [limitations](https://github.com/camilamacedo86/operator-controller/blob/e7ee00514b34d2e75975cb2d63775a0595fbdc48/docs/project/olmv1_limitations.md).
